dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -527,6 +529,16 @@ def login
     I18n.t('auth.unable_to_connect_provider', provider: I18n.t("auth.#{auth_option.credential_type}"))
   end
 
+  private def check_account_linking_lock
+    user = current_user || find_user_by_credential
+    return unless user
+
+    lock_reason = account_linking_lock_reason(user)
+    return unless lock_reason
+
+    redirect_to edit_user_registration_path, alert: lock_reason
```

Comment:
If the user isn't logged in will this redirect them to the sign-in page? Just need to make sure that it works for users who haven't finished registration yet.

Reply:
I implemented behavior similar to `OmniauthCallbacksController#connect_provider`. What should we do for `#link_account` then?
https://github.com/code-dot-org/code-dot-org/blob/abcfd4d83a47adbbaac93c787454243255ca0b07/dashboard/app/controllers/omniauth_callbacks_controller.rb#L88-L91

Reply:
I think it's fine to either redirect them to the page they came from, or to the sign-in page, as long as the alert is present.